src/Orleans.Core/GrainDirectory/IGrainLocator.cs

```diff
@@ -15,5 +15,10 @@ public interface IGrainLocator
         ValueTask<ActivationAddress> Lookup(GrainId grainId);
 
         bool TryLocalLookup(GrainId grainId, out ActivationAddress addresses);
+
+        void CachePlacementDecision(ActivationAddress address);
+
+        void InvalidateCache(GrainId grainId);
+        bool TryCachedLookup(GrainId grainId, out ActivationAddress address);
```

Comment:
`TryLocalLookup` and `TryCachedLookup` sound very similar...

Reply:
If I remove the 'd' and make it `TryCacheLookup` is that clearer (or less clear)? Alternatively, how about `TryCacheOnlyLookup`?
`TryLocalLookup` really only makes sense for the DHT directory, so I think it's the outlier.

Reply:
I think we could remove `TryLocalLookup` and keep the new `TryCachedLookup`. The `Lookup` method returns a `ValueTask`, so we could do the specific local lookup for the DHT directory in there.